presto-cli/src/main/java/com/facebook/presto/cli/Query.java

```diff
@@ -179,11 +188,14 @@ else if (results.getColumns() == null) {
         return true;
     }
 
-    private void waitForData()
+    private void processInitialStatusUpdates(WarningsPrinter warningsPrinter)
     {
         while (client.isRunning() && (client.currentData().getData() == null)) {
+            warningsPrinter.print(client.currentStatusInfo().getWarnings(), true, false);
             client.advance();
         }
+        List<PrestoWarning> warnings = (client.isRunning() ? client.currentStatusInfo() : client.finalStatusInfo()).getWarnings();
```

Comment:
The inlined conditional expression makes this hard to read. Extract it into a variable or add the `.getWarnings()` to each of the branches.